server/model/monitor.js

```diff
@@ -598,25 +600,12 @@ class Monitor extends BeanModel {
                     } else if (this.type === "json-query") {
                         let data = res.data;
 
-                        // convert data to object
-                        if (typeof data === "string" && res.headers["content-type"] !== "application/json") {
-                            try {
-                                data = JSON.parse(data);
-                            } catch (_) {
-                                // Failed to parse as JSON, just process it as a string
-                            }
-                        }
-
-                        let expression = jsonata(this.jsonPath);
+                        const { status, response } = await evaluateJsonQuery(data, this.jsonPath, this.jsonPathOperator, this.expectedValue);
 
-                        let result = await expression.evaluate(data);
+                        bean.status = status ? UP : DOWN;
+                        bean.msg = `JSON query ${status ? "passes" : "does not pass"} `;
+                        bean.msg += `comparison: ${response} ${this.jsonPathOperator} ${this.expectedValue}.`;
```

Comment:
Otherwise the retry-logic in the try-catch block is not triggered

```suggestion
                        if (status) {
                            bean.status = UP; 
                            bean.msg = `JSON query passes (comparing ${response} ${this.jsonPathOperator} ${this.expectedValue})`
                        } else {
                            throw new Error(`JSON query does not pass (comparing ${response} ${this.jsonPathOperator} ${this.expectedValue})`);
                        }
```